system_test/aest_hard_fork_SUITE.erl

```diff
@@ -445,6 +506,134 @@ new_node_can_mine_on_old_chain_using_new_protocol(Cfg) ->
     aest_nodes:kill_node(new_node3, Cfg),
     ok.
 
+%% New node can sync the old chain from other new node and can start mining
+%% on the top of the old chain. The new node can mine blocks using the old
+%% protocol and include spend transaction in the blocks. The block with the
+%% spend transction can be synced among other new nodes.
+new_node_can_mine_spend_tx_on_old_chain_using_old_protocol(Cfg) ->
+    {_, {_Saver, SavedCfg}} = proplists:lookup(saved_config, Cfg),
+    {_, new_node1} = proplists:lookup(new_node_left_running_with_old_chain, SavedCfg),
+    {_, TopHeight} = proplists:lookup(db_backup_top_height, Cfg),
+    {_, TopHash} = proplists:lookup(db_backup_top_hash, Cfg),
+    %% Sync the chain on new_node3 from new_node1.
+    start_node(new_node3, Cfg),
+    wait_for_height_syncing(TopHeight, [new_node3], {{45000, ms}, {200, blocks}}, Cfg),
+    #{hash := TopHash} = get_block_by_height(new_node3, TopHeight, Cfg),
+    %% Get public key of new_node1 and make sure its balance is 0.
+    PubKey1 = get_public_key(new_node1, Cfg),
+    Balance1 = get_balance(new_node1, PubKey1, Cfg),
+    ?assertEqual(0, Balance1),
+    ct:log("Balance of node ~p with public key ~p is ~p", [new_node1, PubKey1, Balance1]),
+    %% Kill new_node1, it was used just to sync the old chain.
+    aest_nodes:kill_node(new_node1, Cfg),
+    %% Get public key of new_node3 and make sure its balance is 0.
+    PubKey3 = get_public_key(new_node3, Cfg),
+    Balance3 = get_balance(new_node3, PubKey3, Cfg),
+    ?assertEqual(0, Balance3),
+    ct:log("Balance of node ~p with public key ~p is ~p", [new_node3, PubKey3, Balance3]),
+    %% Start mining on new_node3.
+    ok = mock_pow_on_node(new_node3, Cfg),
+    run_erl_cmd_on_node(new_node3, "aec_conductor:start_mining().", "ok", Cfg),
+    %% Check that new_node3 got reward for mining.
+    MinedReward = 100,
+    aest_nodes:wait_for_value({balance, PubKey3, MinedReward}, [new_node3], 10000, Cfg),
+    MinedBalance = get_balance(new_node3, PubKey3, Cfg),
+    ?assert(MinedBalance >= MinedReward),
+    ct:log("Mined balance on ~p with public key ~p is ~p", [new_node3, PubKey3, MinedBalance]),
+    %% Send spend transaction from new_node3 to new_node1.
+    Fee = 5,
+    BalanceToSpend = MinedBalance - Fee,
+    ok = post_spend_tx(new_node3, PubKey1, BalanceToSpend, Fee, Cfg),
+    ct:log("Sent spend tx with balance of ~p from node ~p with public key ~p to node ~p with public key ~p",
+           [BalanceToSpend, new_node3, PubKey3, new_node1, PubKey1]),
+    %% Make sure new_node1 received the balance.
+    aest_nodes:wait_for_value({balance, PubKey1, BalanceToSpend}, [new_node3], 20000, Cfg),
+    ReceivedBalance = get_balance(new_node3, PubKey1, Cfg),
+    ct:log("Balance of node ~p with public key ~p is ~p", [new_node1, PubKey1, ReceivedBalance]),
+    ?assertEqual(BalanceToSpend, ReceivedBalance),
```

Comment:
Would it be possible to assert that block including spend tx has expected protocol version? So this test would be complete.

E.g.
* http://31.13.249.77:3013/v2/account/txs/ak$3fyQbxqu6UJMiJ92A9mPuEqqHV4V7VEfELmQsMzjL3ZFyLXTDFfTvnGGnPq9AKg6fyTMuwZpSPaNFahScNGaeuwTZfnDjQ?tx_types=spend_tx&tx_encoding=json has body with `"block_hash":"bh$2XyRRKjsZRrf7kQgpWRQY7DepYTaum8ohezetwyj1YcDrtA7Dr"`
* http://31.13.249.77:3013/v2/block-by-hash?hash=bh$2XyRRKjsZRrf7kQgpWRQY7DepYTaum8ohezetwyj1YcDrtA7Dr has body with version in it

Reply:
New checks of block that contains the spend transaction are added in 4389749.